projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -996,8 +1129,90 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
                 this.selectedDates = [start, ...this.generateDateRange(start, end)];
             }
         }
+
+        if (excludeDisabledDates) {
+            this.selectedDates = this.selectedDates.filter(d => !this.isDateDisabled(d));
+        }
+
         this._onChangeCallback(this.selectedDates);
     }
+    /**
+     *@hidden
+     */
+    private selectDateFromClient(value: Date) {
+        switch (this.selection) {
+            case 'single':
+            case 'multi':
+                this.selectDate(value);
```

Comment:
Why would the disabled dates be selected in the case of `milti`?

Reply:
The method `selectDateFromClient` is called per each clicked date from the UI. The disabled dates are not clickable, that's why for `single` and `multi` selection we call the API method directly. But if the selected dates are in a range, the disabled dates from that range should no be selected. Please elaborate, if this does not answer you question.

Reply:
So, calling this as an API method will allow me to select disabled dates in the case of `single` and `multiple`, but will not allow me to select them in the case of `range`. It's inconsistent.